app/uk/gov/hmrc/apihubapplications/models/application/ApplicationLenses.scala

```
@@ -152,15 +162,23 @@ object ApplicationLenses {
         application.getSecondaryCredentials.filterNot(_.clientId == clientId)
       )
 
-    def updateSecondaryCredential(credential: Credential): Application = {
-      application.getSecondaryCredentials.indexWhere(_.clientId == credential.clientId) match {
-        case -1 => throw new IllegalArgumentException(
-          s"Application with Id ${application.id.getOrElse("<none>")} does not have a credential with Client Id ${credential.clientId}"
-        )
-        case i => application.setSecondaryCredentials(
-          application.getSecondaryCredentials.updated(i, credential)
+    def updateSecondaryCredential(clientId: String, secret: String): Application = {
+      if (!application.getSecondaryCredentials.exists(_.clientId == clientId)) {
+        throw new IllegalArgumentException(
```

Wouldn't we use NotUpdatedException for this case?

---

The use of IllegalArgumentException was already there although I think I did write that in the first place. We don't expect this to happen ever and this simply forces a fail fast scenario rather than not update anything and appear to succeed.